File: ADT/stack-queue-pblm/1441.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        int stream = 1;
        int i = 0;
        vector<string> result;
        while (i < target.size() && stream <= n) {
            result.push_back("Push");
            if (target[i] == stream) {
                i++;
            } else {
                result.push_back("Pop");
            }
            stream++;
        }
        return result;
    }
};

File: ADT/stack-queue-pblm/1441.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        stream = 1
        i = 0
        result = []
        while i < len(target) and stream <= n:
            result.append("Push")
            if target[i] == stream:
                i += 1
            else:
                result.append("Pop")
            stream += 1
        return result